channeldb/channel_cache.go

```diff
@@ -0,0 +1,50 @@
+package channeldb
+
+// channelCache is an in-memory cache used to improve the performance of
+// ChanUpdatesInHorizon. It caches the chan info and edge policies for a
+// particular channel.
+type channelCache struct {
+	n        int
+	channels map[uint64]ChannelEdge
```

Comment:
Why not a pointer value here? Less GC pressure with a large cache?

Reply:
yes, and because a `ChannelEdge` itself consists of only 3 pointers and so are relatively small